wcomponents-test-lib/src/main/java/com/github/bordertech/wcomponents/test/selenium/ByButton.java

```diff
@@ -0,0 +1,87 @@
+package com.github.bordertech.wcomponents.test.selenium;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.internal.FindsById;
+import org.openqa.selenium.internal.FindsByXPath;
+
+import java.util.List;
+
+/**
+ * Finds a button by the value on the button.
+ */
+public class ByButton extends By {
+
+	/**
+	 * Button value by exact root.
+	 */
+	public static final String XPATH_BUTTON_VALUE_EXACT_ROOT = "//button[text()='%1$s']";
```

Comment:
This is a good point, i have changed the other new By extensions to be specific IE ByButtonText is only the button text (removed the ID as we should just use the By.id which already exists) and have added a ByButtonImage which currently looks at a css style class "wc-icon" which may need changing. Perhaps we could just implement these By extensions on the way to grab element rather than a single class to control all ways.